configs/defs/centos.textproto

```diff
@@ -0,0 +1,1997 @@
+    description:
+      "Mail Transfer Agents (MTA), such as sendmail and Postfix, are used to listen for incoming"
+      "mail and transfer the messages to the appropriate user or mail server. If the system is not"
+      "intended to be a mail server, it is recommended that the MTA be configured to only process"
+      "local mail."
+    rationale:
+      "The software for all Mail Transfer Agents is complex and most have a long history of"
+      "security issues. While it is important to ensure that the system can process local mail"
+      "messages, it is not necessary to have the MTA's daemon listening on a port unless the"
+      "server is intended to be a mail server that receives and processes mail from other systems."
+      "Notes:"
+      "This recommendation is designed around the postfix mail server."
+      "Depending on your environment you may have an alternative MTA installed such as"
+      "sendmail. If this is the case consult the documentation for your installed MTA to"
+      "configure the recommended state."
+    remediation:
+      "Edit /etc/postfix/main.ct and add the following line to the RECEIVING MAIL section. If"
+      "the line already exists, change it to look like the line below:"
+      "inet_interfaces = loopback-only"
+      "Run the following command to restart postfix:"
+      "# systemctl restart postfix"
+    cis_benchmark: {
+      profile_level: 1
+      severity: LOW
+    }
+    scan_instructions:
+      "generic:{"
+      "  check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{single_file:{path:\"/proc/self/net/tcp\"}}"
+      "      content_entry:{"
+      "        match_type: ALL_MATCH_ANY_ORDER"
+      "        match_criteria: {"
+      "          filter_regex: \".*0100007F:0019.*\""
+      "          expected_regex: \".*0100007F:0019.*\""
+      "        }"
+      "      }}"
+      "  }"
+      "  check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{single_file:{path:\"/proc/self/net/tcp6\"}}"
+      "      content_entry:{"
+      "        match_type: ALL_MATCH_ANY_ORDER"
+      "        match_criteria: {"
+      "          filter_regex: \".*00000000000000000000000001000000:0019.*\""
+      "          expected_regex: \".*00000000000000000000000001000000:0019.*\""
+      "        }"
+      "      }}"
+      "  }"
+      "  check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{"
+      "      single_file:{path:\"/proc/self/net/tcp6\"}"
+      "      }"
+      "      files_to_check:{"
+      "      single_file:{path:\"/proc/self/net/tcp\"}"
+      "      }"
+      "      content_entry:{"
+      "        match_type: NONE_MATCH"
+      "        match_criteria: {"
+      "          filter_regex: \".*[0-9A-F]+:0019.*\""
+      "        }"
+      "      }}"
+      "  }"
+      "}"
+  }
+}
+benchmark_configs: {
+  id: "iptables-services-not-installed-with-firewalld"
```

Comment:
The "iptables-services-not-installed-with-firewalld" has "Ensure firewalld is installed" as title. Is this "3.5.1.1 Ensure firewalld is installed (Automated)" or "3.5.1.2 Ensure iptables-services not installed with firewalld (Automated)"?
It looks like we are missing one of these two.

Reply:
The correct title is "Ensure iptables-services not installed with firewalld". 
So for 3.5.1.1 (Ensure firewalld is installed), 3.5.2.1 (Ensure nftables is installed) and 3.5.3.1.1 (Ensure iptables packages are installed) I decided to implement 3.5.3.1.1 because iptables is the default configuration. I have done that because as far as I understood, some firewall benchmarks are in contradiction: those three benchmarks require that a particular firewall interface is installed (e.g. firewalld) but then there are other benchmarks (3.5.3.1.3 Ensure firewalld is either not installed or masked with iptables ) that requires that only one of them is installed or is masked.

Reply:
Do you mind implementing all firewall checks, even if they are in contradiction? Hopefully you can copy paste the most of it.

After we have all implementations, we can just choose which ones to enable using the "reduced" configs (example: https://github.com/google/localtoast/blob/master/configs/reduced/debian_10/vm_image_scanning.textproto). I'd like to have the full set for each of firewalld, iptables, nftables so that we can swap between them if needed.